eclipse-collections/src/main/java/org/eclipse/collections/impl/map/ordered/mutable/OrderedMapAdapter.java

```diff
@@ -0,0 +1,803 @@
+        }
+        return output;
+    }
+
+    @Override
+    public MutableOrderedMap<K, V> takeWhile(Predicate<? super V> predicate)
+    {
+        throw new UnsupportedOperationException(this.getClass().getSimpleName() + ".takeWhile() not implemented yet");
+    }
+
+    @Override
+    public MutableOrderedMap<K, V> drop(int count)
+    {
+        if (count < 0)
+        {
+            throw new IllegalArgumentException("Count must be greater than zero, but was: " + count);
+        }
+
+        MutableOrderedMap<K, V> output = OrderedMapAdapter.adapt(new LinkedHashMap<>(this.size() - Math.min(this.size(), count)));
+        Iterator<Entry<K, V>> iterator = this.delegate.entrySet().iterator();
+        int start = Math.min(count, this.size());
+        if (start == this.size())
+        {
+            return output;
+        }
+        int i = 0;
+        while (iterator.hasNext())
+        {
+            if (i >= start)
+            {
+                Entry<K, V> next = iterator.next();
+                output.put(next.getKey(), next.getValue());
+            }
+            else
+            {
+                iterator.next();
+            }
+            i++;
+        }
+        return output;
+    }
+
+    @Override
+    public MutableOrderedMap<K, V> dropWhile(Predicate<? super V> predicate)
+    {
+        throw new UnsupportedOperationException(this.getClass().getSimpleName() + ".dropWhile() not implemented yet");
+    }
+
+    @Override
+    public PartitionMutableList<V> partitionWhile(Predicate<? super V> predicate)
+    {
+        throw new UnsupportedOperationException(this.getClass().getSimpleName() + ".partitionWhile() not implemented yet");
+    }
+
+    @Override
+    public MutableList<V> distinct()
+    {
+        throw new UnsupportedOperationException(this.getClass().getSimpleName() + ".distinct() not implemented yet");
+    }
+
+    @Override
+    public MutableOrderedMap<K, V> withKeyValue(K key, V value)
+    {
+        this.put(key, value);
+        return this;
+    }
+
+    @Override
+    public MutableOrderedMap<K, V> withAllKeyValues(Iterable<? extends Pair<? extends K, ? extends V>> keyValues)
+    {
+        keyValues.forEach(keyVal -> this.put(keyVal.getOne(), keyVal.getTwo()));
+        return this;
+    }
+
+    @Override
+    public MutableOrderedMap<K, V> withAllKeyValueArguments(Pair<? extends K, ? extends V>... keyValues)
+    {
+        return this.withAllKeyValues(ArrayAdapter.adapt(keyValues));
+    }
+
+    @Override
+    public MutableOrderedMap<K, V> withoutKey(K key)
+    {
+        this.removeKey(key);
+        return this;
+    }
+
+    @Override
+    public MutableOrderedMap<K, V> withoutAllKeys(Iterable<? extends K> keys)
+    {
+        keys.forEach(this::removeKey);
+        return this;
+    }
+
+    @Override
+    public MutableOrderedMap<K, V> asUnmodifiable()
+    {
+        throw new UnsupportedOperationException(this.getClass().getSimpleName() + ".asUnmodifiable() not implemented yet");
```

Comment:
Do we plan to have `UnmodifiableMutableOrderedMap` in the future? I presume it will be after we have our own implementation of `OrderedMap`

Reply:
Sure